updater

```diff
@@ -324,8 +283,69 @@ list_updates_cli () {  #from https://unix.stackexchange.com/a/673436
 
 
 list_updates_gui() { #input: updatable_apps and updatable_files variables
-  LIST="$(generate_yad_list)"
-  #echo "List: ${LIST}EOL"
+  local IFS=$'\n'
+  
+  #If updatable_apps and updatable_files variables empty, set them to the results of the last update-check
+  if [ -z "$updatable_apps" ] && [ -z "$updatable_files" ];then
+    updatable_apps="$(cat "${DIRECTORY}/data/update-status/updatable-apps")"
+    updatable_files="$(cat "${DIRECTORY}/data/update-status/updatable-files")"
+  fi
+  
+  for app in $updatable_apps ;do #generate a yad list for every updatable app
+    # get app tooltip
+    read -r 2>/dev/null line <"${DIRECTORY}/update/pi-apps/apps/${app}/description" || line="Description unavailable"
+    if [[ ! -d "${DIRECTORY}/apps/${app}" ]]; then
+      LIST+="TRUE
+${DIRECTORY}/update/pi-apps/apps/${app}/icon-24.png
+$app (new app)
+app:$app
+$line"$'\n'
+    elif will_reinstall "$app"; then
+      LIST+="TRUE
+${DIRECTORY}/update/pi-apps/apps/${app}/icon-24.png
+$app (app$(echo ', <b>will be reinstalled</b>'))
+app:$app
+$line"$'\n'
+    else # app is not new or getting re-installed, add to compressed update list
+      compressed_update+=$'\n'"app:${app}"
+    fi
+  done
+  
+  for file in $updatable_files ;do #generate a yad list for every updatable file
+    
+    #determine mimetype of updatable_apps file to display an informative icon in the list
+    if [ "$(file -b --mime-type "${DIRECTORY}/${file}")" == 'text/x-shellscript' ];then
+      #if updatable_apps file in question is a shell script, then display shellscript icon.
+      mimeicon="${DIRECTORY}/icons/shellscript.png"
+      mimetype='script'
+    elif [[ "${DIRECTORY}/${file}" == *.png ]] || [[ "${DIRECTORY}/${file}" == *.svg ]];then
+      mimeicon="${DIRECTORY}/icons/image.png"
+      mimetype='image'
+    else
+      #otherwise display txt icon.
+      mimeicon="${DIRECTORY}/icons/txt.png"
+      mimetype='file'
+    fi
+    
+    LIST+="TRUE
+${mimeicon}
+$file ($mimetype)
+file:$file
+No description"$'\n'
```

Comment:
I think it would be better to not display any tooltip if no useful information is contained in it.
I just tested yad with an empty tooltip column, and it correctly avoided displaying an empty tooltip box.